File: WakeUpIntel.java
import java.net.InetAddress;
import java.util.Properties;
import javax.net.ssl.SSLException;
import com.intel.amt.sdk.*;
import com.intel.amt.api.*;
import java.util.Scanner;

/**
 * This class will send a wake-on-LAN command to an Intel AMT device.
 * 
 * @author Michael Fink @imafink
 * @version 1.0
 */

public class WakeUpSystemIntel {
    public static void wakeSystem(String ipAddress, String username, String password) {
        // Initialize AMT SDK properties
        Properties properties = new Properties();
        properties.setProperty("applicationName", "WakeUpSystem");
        properties.setProperty("useTls", "true");

        // Create connection details object
        ConnectionDetails connectionDetails = new ConnectionDetails(
                ipAddress, 16992, username, password, KvmUtil.getAmtPasswordCharset());

        // Create AMT session object
        AmtSession amtSession = new AmtSession(connectionDetails, properties);

        // Send power-on command to AMT device
        try {
            // Establish AMT session
            amtSession.connect();

            // Create power-on request object
            PowerUpRequest powerUpRequest = new PowerUpRequest(
                    PowerUpType.WOL, NetworkSettings.DEFAULT_WAKE_ON_LAN_PACKET);

            // Send power-on request to AMT device
            amtSession.getServiceProcessorClient().powerUp(powerUpRequest);

            // Disconnect AMT session
            amtSession.disconnect();

            System.out.println("Wake-on-LAN command sent to " + ipAddress);
        } catch (Exception e) {
            System.err.println("Error sending wake-on-LAN command: " + e.getMessage());
        }
    }

    public static void main(String[] args) {
        String ipAddress = "";
        String username = "";
        String password = "";
        boolean repeat = true;

        // General print block
        System.out.println("Welcome to the Intel Wake-Up Program. Please follow the instructions below.")
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter Username: ");
        username = scan.nextLine();
        System.out.println("Enter Password: ");
        password = scan.nextLine();
        System.out.println("Type exit to exit the program.");

        // Loop for multiple machines
        while (repeat) {
            System.out.println("Enter IP Address: ");
            ipAddress = scan.nextLine();
            if (ipAddress.equals("exit") || ipAddress.equals("Exit") || ipAddress.equals("EXIT")) {
                repeat = false;
            } else {
                wakeSystem(ipAddress, username, password);
            }
        }
        System.out.println("Thank you for using the Intel Wake-Up Program.");
        scan.close();
    }
}

File: WakeUpIntel.cpp
#include <iostream>
#include <string>
#include "AMT.h" // Include the Intel AMT SDK header file

using namespace std;

void wakeUpSystem(const string& ipAddress, const string& username, const string& password) {
    try {
        // Initialize AMT SDK
        AMT::AMTSDKInitializer amtSDKInitializer;

        // Create connection details object
        AMT::AMTConnDetails connDetails(ipAddress, username, password);

        // Create AMT session
        AMT::AMTSession amtSession(connDetails);

        // Connect to AMT device
        amtSession.connect();

        // Send power-on command to AMT device
        AMT::AMTStatus status = amtSession.powerUp(AMT::AMTPowerUpType::PowerUp_WOL);

        // Disconnect from AMT device
        amtSession.disconnect();

        if (status == AMT::AMTStatus::OK) {
            cout << "Wake-on-LAN command sent to " << ipAddress << endl;
        }
        else {
            cout << "Failed to send wake-on-LAN command: " << AMT::AMTStatusToString(status) << endl;
        }
    }
    catch (exception& e) {
        cerr << "Error: " << e.what() << endl;
    }
}

int main() {
    string ipAddress;
    string username;
    string password;

    // Get user input
    cout << "Enter IP Address: ";
    getline(cin, ipAddress);
    cout << "Enter Username: ";
    getline(cin, username);
    cout << "Enter Password: ";
    getline(cin, password);

    // Call the wakeUpSystem function
    wakeUpSystem(ipAddress, username, password);

    return 0;
}